sktime/forecasting/base/adapters/_neuralforecast.py

```diff
@@ -189,15 +191,45 @@ def _fit(
         if not fh.is_all_out_of_sample(cutoff=self.cutoff):
             raise NotImplementedError("in-sample prediction is currently not supported")
 
-        if self.freq == "auto" and fh.freq is None:
-            # when freq cannot be interpreted from ForecastingHorizon
-            raise ValueError(
-                f"Error in {self.__class__.__name__}, "
-                f"could not interpret freq, "
-                f"try passing freq in model initialization"
-            )
-
-        self._freq = fh.freq if self.freq == "auto" else self.freq
+        # A. freq is given {use this}
+        # B. freq is auto
+        #     B1. freq is infered from fh {use this}
+        #     B2. freq is not infered from fh
+        #         B2.1. y is date-like {raise exception}
+        #         B2.2. y is not date-like
+        #             B2.2.1 uniform data {use diff in time}
+        #             B2.2.2 missing data {raise exception}
+
+        if self.freq != "auto":
+            # A
+            self._freq = self.freq
+        else:
+            # B
+            if fh.freq:
+                # B1
+                self._freq = fh.freq
+            else:
+                # B2
+                if isinstance(y.index, (pandas.PeriodIndex, pandas.DatetimeIndex)):
+                    # B2.1
+                    raise ValueError(
+                        f"Error in {self.__class__.__name__}, "
+                        f"could not interpret freq, "
+                        f"try passing freq in model initialization"
+                    )
+                else:
+                    # B2.2
+                    diffs = np.unique(np.diff(y.index))
+                    if diffs.shape[0] > 1:
+                        # B2.2.1
+                        raise ValueError(
+                            f"Error in {self.__class__.__name__}, "
+                            f"could not interpret freq, "
+                            f"try passing integer freq in model initialization"
+                        )
+                    else:
+                        # B2.2.2
+                        self._freq = int(diffs[-1])
```

Comment:
Question: can it be non-integer?

Reply:
no, `neuralforecast` raises error for other than int

Reply:
I am asking can `diffs[-1]` be anything but an integer based on the if-else checks?

Reply:
Technically diffs should have integers only because if not they are early stopped at `self._check_X_y(X=X, y=y)`. But the integers are `numpy.int64` which are not supported by neuralforecast and need to be converted to python `int`

Reply:
Okay. In that case, can you please add this as a comment? I'm sure I'll get confused every time I see this line in future.